.gitignore

```diff
@@ -0,0 +1,12 @@
+# Temporary files.
+.#*
+
+# Log files.
+*.log
+
+# Don't push node_modules or dist - we want the build server to install and build these.
```

Comment:
While this was relevant for the previous version when balenaEngine didn't take note of .dockerignore (or not specifically atleast), we should put this warning over on the `.dockerignore` side. So folks know why we want them to include **node_modules** in **.dockerignore**

Reply:
I see this does take heavy inspiration from https://github.com/balena-io-examples/balena-node-hello-world/, something to remember is it is a 4-year-old repo and we should update it too next FWIW

Reply:
Good point, done.